metagraph-plugin/metagraph_katana/tests/test_algorithms.py

```diff
@@ -0,0 +1,247 @@
+    BetweennessCentralityStatistics,
+    BfsStatistics,
+    ConnectedComponentsStatistics,
+    IndependentSetPlan,
+    IndependentSetStatistics,
+    JaccardPlan,
+    JaccardStatistics,
+    KCoreStatistics,
+    KTrussStatistics,
+    LouvainClusteringStatistics,
+    PagerankStatistics,
+    SsspStatistics,
+    TriangleCountPlan,
+    betweenness_centrality,
+    bfs,
+    bfs_assert_valid,
+    connected_components,
+    connected_components_assert_valid,
+    find_edge_sorted_by_dest,
+    independent_set,
+    independent_set_assert_valid,
+    jaccard,
+    jaccard_assert_valid,
+    k_core,
+    k_core_assert_valid,
+    k_truss,
+    k_truss_assert_valid,
+    local_clustering_coefficient,
+    louvain_clustering,
+    louvain_clustering_assert_valid,
+    pagerank,
+    pagerank_assert_valid,
+    sort_all_edges_by_dest,
+    sort_nodes_by_degree,
+    sssp,
+    sssp_assert_valid,
+    subgraph_extraction,
+    triangle_count,
+)
+
+# directed graph
+@pytest.fixture(autouse=True)
+def kg_from_nx_di_8_12(nx_weighted_directed_8_12):
+    pg_test_case = mg.translate(nx_weighted_directed_8_12, mg.wrappers.Graph.KatanaGraph)
+    return pg_test_case
+
+
+# undirected graph
+@pytest.fixture(autouse=True)
+def kg_from_nx_ud_8_12(nx_weighted_undirected_8_12):
+    pg_test_case = mg.translate(nx_weighted_undirected_8_12, mg.wrappers.Graph.KatanaGraph)
+    return pg_test_case
+
+
+# breadth-first search, both the katana-version and the networkx version are implemented
+# mg.algos.traversal.bfs_iter(graph: Graph, source_node: NodeID, depth_limit: int = - 1) → Vector
+# results of the 2 versions are compared
+def test_bfs(nx_weighted_directed_8_12, kg_from_nx_di_8_12):
+    bfs1_nx = mg.algos.traversal.bfs_iter(nx_weighted_directed_8_12, 0)
+    bfs2_nx = mg.algos.traversal.bfs_iter(nx_weighted_directed_8_12, 2)
+    bfs1_kg =  mg.algos.traversal.bfs_iter(kg_from_nx_di_8_12, 0, 2**30 - 1)
+    bfs2_kg = mg.algos.traversal.bfs_iter(kg_from_nx_di_8_12, 2, 2**30 - 1)
+    assert bfs1_kg.tolist() == bfs1_nx.tolist()
+    assert bfs2_kg.tolist() == bfs2_nx.tolist()
+    assert bfs1_kg.tolist() == [0, 1, 3, 4, 7]
+    assert bfs2_kg.tolist() == [2, 4, 5, 6, 7]
+    # additional tests in katana, feel free to comment this part
+    node_schema = kg_from_nx_di_8_12.value.node_schema()
+    assert node_schema[len(node_schema)-2].name == 'bfs_prop_start_from_0'
+    assert node_schema[len(node_schema)-1].name == 'bfs_prop_start_from_2'
+    bfs_assert_valid(kg_from_nx_di_8_12.value, 0, 'bfs_prop_start_from_0')
+    bfs_assert_valid(kg_from_nx_di_8_12.value, 2, 'bfs_prop_start_from_2')
+    verify_bfs(kg_from_nx_di_8_12.value, 0, len(node_schema)-2)
+    verify_bfs(kg_from_nx_di_8_12.value, 2, len(node_schema)-1)
+    stats1 = BfsStatistics(kg_from_nx_di_8_12.value, 'bfs_prop_start_from_0')
+    stats2 = BfsStatistics(kg_from_nx_di_8_12.value, 'bfs_prop_start_from_2')
+    assert stats1.n_reached_nodes == 5
+    assert stats2.n_reached_nodes == 5
```

Comment:
I don't think these tests do anything that isn't already tested by the main test harness. `kg_from_nx_di_8_12.value` is a PropertyGraph right?

Reply:
Yes.